exercises/allergies/example.cpp

```diff
@@ -0,0 +1,46 @@
+#include "allergies.h"
+
+#include <string>
+#include <map>
+#include <vector>
+#include <algorithm>
+
+namespace allergies
+{
+
+allergy_test::allergy_test(const int test_result)
+    : result(test_result)
+    , allergens({
+    	{"eggs", 1},
+	{"peanuts", 2},
+	{"shellfish", 4},
+	{"strawberries", 8},
+	{"tomatoes", 16},
+	{"chocolate", 32},
+    	{"pollen", 64},
+	{"cats", 128}
+   })
+{} 
+
+bool allergy_test::is_allergic_to(std::string const& allergen) const
+{
+    int allergen_value = allergens.at(allergen);
+    return (result & allergen_value) == allergen_value;
+}
+
+std::vector<std::string> allergy_test::get_allergies() const
+{
+    std::vector<std::string> allergy_list({"eggs", "peanuts", "shellfish", "strawberries",
+		    			   "tomatoes", "chocolate", "pollen", "cats"});
```

Comment:
Wanted to note re: this one in particular, this happened because of trying to shoehorn this into a vector when it shouldn't be. Problem is, std::map is sorted by key, so when iterating over the map's keys to produce the allergen vector, said vector comes back in a different order than the test cases. AFAIK, solving this in a semi-reasonable manner while sticking with vectors would require either normalizing the vector in each test case or moving to boost::bi_map. 

One more reason to prefer using a set, no?